packages/support/lib/image-util.js

```diff
@@ -345,16 +318,25 @@ async function getImagesMatches (img1Data, img2Data, options = {}) {
     count: matches.length,
   };
   if (visualize) {
-    const visualization = cv.drawMatches(img1, img2, result1.keyPoints, result2.keyPoints, matches);
+    const goodMatchesVec = new cv.DMatchVector();
+    for (let i = 0; i < matches.length; i++) {
+      goodMatchesVec.push_back(matches[i]);
+    }
+    const visualization = new cv.Mat();
+    const color = new cv.Scalar(0, 255, 0, 255);
+    cv.drawMatches(img1, result1.keyPoints, img2, result2.keyPoints, goodMatchesVec, visualization, color);
     highlightRegion(visualization, rect1);
     highlightRegion(visualization, {
       x: img1.cols + rect2.x,
       y: rect2.y,
       width: rect2.width,
       height: rect2.height
     });
-    result.visualization = await cv.imencodeAsync('.png', visualization);
+    result.visualization = await jimpImgFromCvMat(visualization).getBufferAsync(Jimp.MIME_PNG);
   }
+
+  img1.delete(); img2.delete(); detector.delete(); result1.keyPoints.delete(); result1.descriptor.delete();
```

Comment:
I assume this stuff must be moved to `finally` section. Otherwise we might get quite a few resource leaks if this function does not reach to its end successfully

Reply:
same below

Reply:
there will potentially be lots of nested try/finallys then, as many of these objects are not defined and used in the same parts of functions. i'll see what i can do though.